test/ClangImporter/indirect_fields.swift

```diff
@@ -0,0 +1,19 @@
+// RUN: %target-swift-frontend -typecheck -sdk "" -I %t -I %S/Inputs/custom-modules %s -verify
+
+import IndirectFields
+
+func build_struct(a: Int32, c: Int32, d: Int32) -> StructWithIndirectField {
+    return StructWithIndirectField(__Anonymous_field0: .init(a: a), c: c, d: d)
```

Comment:
That's not bad, but having a mix of named and nameless arguments is weird. I suppose we could try to trust that most C cases like this will put the transparent struct/union at the front or back of the structure?

@milseman, @DougGregor, thoughts?